### PR TITLE
Dispatch events before and after applying batch action to an object

### PR DESCRIPTION
**Pull request by szymach**

### CODE REVIEW

File: Admin/CRUD/GenericBatchElement.php [hunk 1]
```diff
@@ -27,6 +27,8 @@ public function getRoute()
      */
     public function configureOptions(OptionsResolver $resolver)
     {
+        $resolver->setDefault('abort_if_skipped', false);
+        $resolver->setAllowedTypes('abort_if_skipped', 'bool');
```
Comment (chives) [inline Admin/CRUD/GenericBatchElement.php]: I think this is unnecessary duplication. If user can create event handler which can decide to skip applying action to one object or completely abort the whole batch, then why does he/she need to overwrite that decision in admin element's options? Maybe there's some use case i can't imagine?

File: Event/BatchEvent.php [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public function getSkipElement()
+    {
+        return $this->skipElement;
+    }
+
+    /**
+     * @param boolean $skipElement
+     */
+    public function setSkipElement($skipElement)
```
Comment (chives) [inline Event/BatchEvent.php]: `skip($skip)` should be enough

File: Event/BatchEvent.php [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public function getObject()
+    {
+        return $this->object;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getSkipElement()
```
Comment (chives) [inline Event/BatchEvent.php]: `shouldSkip()` would be more accurate imho. skipping and aborting only make sense before applying action, so i'm wondering if we should create separate event class for "before" event which will extend the event class for "after" event with skipping and aborting feature

File: Event/BatchEvent.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+     */
+    public function setSkipElement($skipElement)
+    {
+        $this->skipElement = $skipElement;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getAbort()
```
Comment (chives) [inline Event/BatchEvent.php]: `shouldAbort()` would be more accurate imho

File: Event/BatchEvent.php [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        return $this->abort;
+    }
+
+    /**
+     * @param boolean $abort
+     */
+    public function setAbort($abort)
```
Comment (chives) [inline Event/BatchEvent.php]: `abort($abort)` should be enough

File: Admin/CRUD/Context/Request/BatchFormValidRequestHandler.php [hunk 6]
```diff
@@ -148,4 +169,35 @@ private function setWarningMessage()
             $this->flashMessages->warning('messages.batch.no_elements');
         }
     }
+
+    /**
+     * @param BatchEvent $event
+     * @return boolean
+     */
+    private function checkIfAbort($event)
```
Comment (chives) [inline Admin/CRUD/Context/Request/BatchFormValidRequestHandler.php]: `shouldAbort(BatchEvent $event)` would be more accurate

File: Admin/CRUD/Context/Request/BatchFormValidRequestHandler.php [hunk 7]
```diff
@@ -148,4 +169,35 @@ private function setWarningMessage()
...
+        }
+
+        return $event->getAbort();
+    }
+
+    /**
+     * @param BatchEvent $event
+     * @return boolean
+     */
+    private function checkIfSkip($event)
```
Comment (chives) [inline Admin/CRUD/Context/Request/BatchFormValidRequestHandler.php]: `shouldSkip(BatchEvent $event)` would be more accurate

File: Admin/CRUD/GenericBatchElement.php (hunk 1 above)
Comment (szymach) [reply Admin/CRUD/GenericBatchElement.php]: This is what you have requested in #298, I can remove it no problem.

Comment (chives) [reply Admin/CRUD/GenericBatchElement.php]: Now i see how bad idea it was

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
@fsi-open

**Comment by szymach** [comment]
Do we wait for @JarekW ?

**Comment by szymach** [comment]
@JarekW

**Pull request merged**